### ISSUE TITLE
python3: Invalid sequence files for non-ascii sequences

### ISSUE THREAD

**Issue by mdehoon** [biopython/biopython#3318]
This is a leftover from the Python2 -> Python3 transition.

The `_data` attribute of a `Seq` object is a `str`.
On Python2, this was an array of bytes (similar to a `bytes` object in python3).
On Python3, this is now a Unicode string.
This means that in principle users can now store non-ASCII characters in a `Seq` object.
However, this is not really useful. We cannot transcribe, translate, complement, or reverse-complement non-ASCII characters, and storing a sequence with non-ASCII characters will result in invalid files.
For example,
```python
>>> from Bio.Seq import Seq
>>> from Bio.SeqRecord import SeqRecord
>>> from Bio import SeqIO
>>> s = Seq("αβγδε")
>>> r = SeqRecord(s, id='chr1')
>>> SeqIO.write([r], "test.fa", 'fasta')
1
```
```bash
$ faSize test.fa
Invalid fasta format: sequence size == 0 for element chr1
0 bases (0 N's 0 real 0 upper 0 lower) in 1 sequences in 1 files
%100.00 masked total, %100.00 masked real

$ cat test.bed
chr1	1	4

$ bedtools getfasta -fi test.fa -bed test.bed
index file test.fa.fai not found, generating...
>chr1:1-4
?β
```

I think then that the `_data` attribute, now that Python2 is out of the way, should be a `bytes` object (and a `bytearray` object in the case of `MutableSeq`). This has the added advantage that file parsing can be significantly faster as no conversion to a `str` will be needed to parse the file.

**Comment by mdehoon** [comment]
Note also that on Python2, Seq objects with unicode characters were broken:
```python
>>> from Bio.Seq import Seq
>>> s = Seq("ACGT")
>>> len(s)
4
>>> s = Seq("àçģţ")
>>> len(s)
8
>>> s = Seq("アイウエオ")
>>> len(s)
15
```

**Comment by peterjc** [comment]
The current length behaviour is broken in this case, given:

```python
>>> len("àçģţ")
4
>>> len("アイウエオ")
5
```

There is currently no good use case for unicode in sequences (although we can imaging an extension defining a character for each amino acid, or even modified nucleotide bases). I am not aware of any file format which deliberately supports unicode.

I agree that using bytes internally for sequences has a lot of positives (although we would need to rework many of our parsers to work in binary mode to see that potential speedup).

**Comment by mdehoon** [comment]
Note also that `array`s with typecode `"u"`, which are being used in `MutableSeq` in the `_data` attribute to store the sequence as unicode characters, have been deprecated since Python 3.3.

**Comment by mdehoon** [comment]
Fixed by #3333, closing.
